From 13dbbf18c1921a3df760982b4648d2513917c764 Mon Sep 17 00:00:00 2001
From: Andre Przywara <andre.przywara@arm.com>
Date: Fri, 16 Nov 2018 01:38:32 +0000
Subject: [PATCH] sunxi: H6: Add basic Eachlink H6 Mini support

The Eachlink H6 Mini is a modestly priced TV box, using the Allwinner H6
SoC. It comes with 3GB of DRAM and 32GB of eMMC in the typical TV box
enclosure.

This adds a basic device tree and defconfig for it.

It contrast to the other supported H6 boards the H6 Mini uses DDR3 DRAM
chips (not LPDDR3), which require a different DRAM controller setup.

Signed-off-by: Andre Przywara <andre.przywara@arm.com>
---
 arch/arm/dts/Makefile                       |   1 +
 arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts | 116 ++++++++++++++++++++
 board/sunxi/MAINTAINERS                     |   6 +
 configs/eachlink_h6_mini_defconfig          |  17 +++
 4 files changed, 140 insertions(+)
 create mode 100644 arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts
 create mode 100644 configs/eachlink_h6_mini_defconfig

diff -Naur u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/arch/arm/dts/Makefile u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/arch/arm/dts/Makefile
--- u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/arch/arm/dts/Makefile	2021-01-09 14:29:18.709999548 +0100
+++ u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/arch/arm/dts/Makefile	2021-01-09 14:31:03.239999545 +0100
@@ -606,6 +606,7 @@
 	sun50i-h5-orangepi-zero-plus2.dtb
 dtb-$(CONFIG_MACH_SUN50I_H6) += \
 	sun50i-h6-beelink-gs1.dtb \
+	sun50i-h6-eachlink-h6-mini.dtb \
 	sun50i-h6-orangepi-3.dtb \
 	sun50i-h6-orangepi-lite2.dtb \
 	sun50i-h6-orangepi-one-plus.dtb \
diff -Naur u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts
--- u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts	2021-01-09 14:30:12.246666213 +0100
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: (GPL-2.0+ or MIT)
+/*
+ * Copyright (c) 2018 Arm Ltd.
+ * based on work by:
+ *   Copyright (c) 2017 Icenowy Zheng <icenowy@aosc.io>
+ */
+
+/dts-v1/;
+
+#include "sun50i-h6.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Eachlink H6 Mini";
+	compatible = "eachlink,h6-mini", "allwinner,sun50i-h6";
+
+	aliases {
+		serial0 = &uart0;
+		ethernet0 = &emac;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	connector {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi_con_in: endpoint {
+				remote-endpoint = <&hdmi_out_con>;
+			};
+		};
+	};
+
+	reg_vcc3v3: vcc3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	reg_vcc5v: vcc5v {
+		/* supply from the micro-USB DC jack */
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-5v";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		regulator-always-on;
+	};
+};
+
+&de {
+	status = "okay";
+};
+
+&hdmi {
+	status = "okay";
+};
+
+&hdmi_out {
+	hdmi_out_con: endpoint {
+		remote-endpoint = <&hdmi_con_in>;
+	};
+};
+
+&ehci0 {
+	phys = <&usb2phy 0>;
+	status = "okay";
+};
+
+&ehci3 {
+	status = "okay";
+};
+
+&mmc0 {
+	vmmc-supply = <&reg_vcc3v3>;
+	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&mmc2 {
+	vmmc-supply = <&reg_vcc3v3>;
+	vqmmc-supply = <&reg_vcc3v3>;
+	non-removable;
+	cap-mmc-hw-reset;
+	bus-width = <8>;
+	status = "okay";
+};
+
+&ohci0 {
+	phys = <&usb2phy 0>;
+	status = "okay";
+};
+
+&ohci3 {
+	status = "okay";
+};
+
+&uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_ph_pins>;
+	status = "okay";
+};
+
+&usb2otg {
+	dr_mode = "host";
+	status = "okay";
+};
+
+&usb2phy {
+	usb0_vbus-supply = <&reg_vcc5v>;
+	status = "okay";
+};
diff -Naur u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/board/sunxi/MAINTAINERS u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/board/sunxi/MAINTAINERS
--- u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/board/sunxi/MAINTAINERS	2021-01-09 14:29:18.709999548 +0100
+++ u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/board/sunxi/MAINTAINERS	2021-01-09 14:30:12.246666213 +0100
@@ -206,6 +206,12 @@
 S:	Maintained
 F:	configs/Cubietruck_plus_defconfig
 
+EACHLINK H6 MINI
+M:	Andre Przywara <andre.przywara@arm.com>
+S:	Maintained
+F:	configs/eachlink_h6_mini_defconfig
+F:	arch/arm/dts/sun50i-h6-eachlink-h6-mini.dts
+
 EMLID NEUTIS N5 DEV BOARD
 M:	Aleksandr Aleksandrov <aleksandr.aleksandrov@emlid.com>
 S:	Maintained
diff -Naur u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/configs/eachlink_h6_mini_defconfig u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/configs/eachlink_h6_mini_defconfig
--- u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-old/configs/eachlink_h6_mini_defconfig	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-sunxi-344e0d6665e53b116a5f422a9f6fae83af786689-new/configs/eachlink_h6_mini_defconfig	2021-01-09 14:30:12.246666213 +0100
@@ -0,0 +1,17 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_NR_DRAM_BANKS=1
+CONFIG_SPL=y
+CONFIG_MACH_SUN50I_H6=y
+CONFIG_DRAM_CLK=660
+CONFIG_DRAM_ZQ=3881979
+CONFIG_SUNXI_DRAM_H6_DDR3_1333=y
+CONFIG_MMC0_CD_PIN="PF6"
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
+# CONFIG_PSCI_RESET is not set
+# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
+CONFIG_SPL_TEXT_BASE=0x20060
+# CONFIG_CMD_FLASH is not set
+# CONFIG_SPL_DOS_PARTITION is not set
+# CONFIG_SPL_EFI_PARTITION is not set
+CONFIG_DEFAULT_DEVICE_TREE="sun50i-h6-eachlink-h6-mini"
